gmso/core/forcefield.py

```diff
@@ -77,6 +82,69 @@ def atom_class_groups(self):
                 atomclass_dict[atom_type.atomclass] = atomclass_group
         return atomclass_dict
 
+    def group_by_expression(self):
+        """Create ForceField member groups having same expressions"""
+        atom_type_groups = self._group_by_expression(_type='AtomType')
+        bond_type_groups = self._group_by_expression(_type='BondType')
+        angle_type_groups = self._group_by_expression(_type='AngleType')
+        dihedral_type_groups = self._group_by_expression(_type='DihedralType')
+        return {
+            'atom_types': atom_type_groups,
+            'bond_types': bond_type_groups,
+            'angle_types': angle_type_groups,
+            'dihedral_types': dihedral_type_groups
+        }
+
+    def _group_by_expression(self, _type):
+        type_groups = {
+                       'AtomType': self.atom_types,
+                       'BondType': self.bond_types,
+                       'AngleType': self.angle_types,
+                       'DihedralType': self.dihedral_types
+                    }
+        grouped_coll = {}
+        this_type = type_groups.get(_type, [])
+        for type_name in this_type:
+            this_expression = str(this_type[type_name].expression)
+            grouped_coll[this_expression] = grouped_coll.get(this_expression, [])
+            grouped_coll[this_expression].append(this_type[type_name])
+        return grouped_coll
+
+    def to_xml(self, filename=None, overwrite=True):
+        """Save this forcefield as a topology XML file
```

Comment:
Should this docstring replace "topology" with "gmso"? Unclear to me which references are supposed to be the (former) package name vs. actually meaning "topology."
```suggestion
        """Save this forcefield as a topology XML file
```